src/cli/moz-cache.js

```diff
@@ -0,0 +1,217 @@
+        ['emulator_url', 'dolphin-512'],
+        ['g_emulator_url', 'emulator.xml'],
+        ['g_emulator_url', 'emulator-jb.xml'],
+        ['g_emulator_url', 'emulator-kk.xml'],
+        ['g_emulator_url', 'emulator-l.xml']
+    ];
+
+    var scheduler = new Scheduler({
+        // for use with taskclusterProxy
+        baseUrl: 'taskcluster/scheduler/v1'
+    });
+    var tasks = [];
+
+    for (var url of clones) {
+        tasks.push(generateCloneTaskDefinition(url));
+    }
+
+    for (var emulator of emulators) {
+        tasks.push(generateRepoCacheTaskDefinition(emulator[1], emulator[0]));
+    }
+
+
+    let {graphId, graph} = generateCacheGraph(tasks);
+    console.log(`Creating cache graph '${graphId}'`);
+    console.log(JSON.stringify(graph));
+
+    let result = await scheduler.createTaskGraph(graphId, graph);
+    console.log(JSON.stringify(result));
+}
+
+
+// To configure, define in your environment:
+//   OWNER_EMAIL
+//   TASKCLUSTER_ACCESS_TOKEN
+//   TASKCLUSTER_CLIENT_ID
+//
+// Notes:
+// * Only supports a workerType with permacreds at this time.
+
+function generateCacheGraph(tasks) {
+  let graphId = slugid.nice();
+
+  let client_id = '';
+  if (process.env.TASK_ID) {  // we're running in docker-worker in TaskCluster prod
+      client_id = 'hook-id:taskcluster/tc-vcs-refresh';
+  } else {
+      client_id = 'client-id:' + process.env.TASKCLUSTER_CLIENT_ID;
+  } // TODO maybe should just totally bail at this point
+
+  let graph = {
+      metadata: {
+        name:         'Clone Cache Graph',
+        description:  'Creates cached copies of repositories to use with taskcluster-vcs',
+        owner:        process.env.OWNER_EMAIL || "selena@mozilla.com",
+        source:       'https://github.com/taskcluster/taskcluster-vcs'
+      },
+      scopes: ['assume:' + client_id]
+  }
+
+  graph.tasks = tasks.map((task) => {
+    return {
+      taskId: slugid.nice(),
+      task: task
+    }
+  })
+
+  return {graphId, graph};
+}
+
+function setDeadline(date, hoursFromNow) {
+    date.setHours(date.getHours() + hoursFromNow);
+    return date;
+}
+
+function generateCloneTaskDefinition(repo) {
+
+    var date = new Date();
+    var deadline = setDeadline(new Date(date.getTime()), 4);
+    var params = ['create-clone-cache', '--force-clone', '--upload', '--proxy', repo]
+
+    var task = {
+      provisionerId: 'aws-provisioner-v1',
+      workerType: 'gaia',
+      created: date,
+      deadline: deadline,
+      scopes: ['queue:create-artifact:*', 'index:insert-task:tc-vcs.v1.clones.*'],
+      payload: {
+        image: 'taskcluster/taskcluster-vcs:2.3.22',
+        command: params,
+        maxRunTime: 3600,
+        features: {
+          taskclusterProxy: true
+        },
+        env: {
+          DEBUG: '*'
+        }
+      },
+      metadata: {
+        name: `cache ${repo}`,
+        description: params.join(' '),
+        owner: process.env.OWNER_EMAIL || "selena@mozilla.com",
```

Comment:
if really wanted to, instead of checking the environment, or defaulting to you, this could just be a param we pass into moz-cache